### DISCUSSION TITLE
Is require('wdio-electron-service/main'); in main/preload still necessary for electron test

### DISCUSSION THREAD

**Discussion by mulgurul** [webdriverio/webdriverio#11576] [General]
This is probably most a question to Christian after watching the quick tutorial on setting up Electron e2e test.

Before I read several places, and saw in example code that it was required to call require('wdio-electron-service/[main|preload]'); in main and preload for testing  if in development mode. 

But in the youtube tut. you don't mention this and testing seems to work fine.
So is this a requirement of the past?

Thanks.

**Comment by christian-bromann** [comment]
> So is this a requirement of the past?

It is still a requirement if you like to tap into the application runtime for [mocking](https://webdriver.io/docs/wdio-electron-service#mocking-electron-apis) or just executing scripts. We will soon release an interface that allows you to mock as following:

```ts
const mock = await browser.electron.mock('dialog', '...')
// click on a button that causes a dialog to pop up
await expect(mock).toHaveBeenCalledTimes(1)
```

If you don't need any of that you don't need to import the scripts.

**Reply by 255kb** [reply]
Hi, if we don't need the mocks, is there a way to disable the multiple checks for the `wdio-electron-service/preload` presence during the tests? It generates many warnings/errors:

```
Find more information at https://webdriver.io/docs/desktop-testing/electron#api-configuration
  (Session info: chrome=122.0.6261.70)
2024-02-29T15:08:30.366Z ERROR webdriver: Request failed with status 500 due to javascript error: javascript error: Electron context bridge not available! Did you import the service hook scripts into your application via e.g. `import('wdio-electron-service/main')` and `import('wdio-electron-service/preload')`?
```
